sklearn/tests/test_multioutput.py

```diff
@@ -491,3 +421,89 @@ def test_classifier_chain_vs_independent_models():
 
     assert_greater(jaccard_similarity_score(Y_test, Y_pred_chain),
                    jaccard_similarity_score(Y_test, Y_pred_ovr))
+
+
+def test_base_chain_fit_and_predict():
+    # Fit base chain and verify predict performance
+    X, Y = generate_multilabel_dataset_with_correlations()
+    chains = [RegressorChain(LinearRegression()),
+              ClassifierChain(LogisticRegression())]
+    [chain.fit(X, Y) for chain in chains]
+
+    for i in range(0, 2):
+        Y_pred = chains[i].predict(X)
+        assert_equal(Y_pred.shape, Y.shape)
+
+        assert_equal([c.coef_.size for c in chains[i].estimators_],
+                     list(range(X.shape[1], X.shape[1] + Y.shape[1])))
+
+    Y_prob = chains[1].predict_proba(X)
+    Y_binary = (Y_prob >= .5)
+    assert_array_equal(Y_binary, Y_pred)
+
+    assert isinstance(chains[0], RegressorMixin)
+    assert isinstance(chains[1], ClassifierMixin)
+
+
+def test_base_chain_fit_and_predict_with_sparse_data_and_cv():
+    # Fit base chain with sparse data cross_val_predict
+    X, Y = generate_multilabel_dataset_with_correlations()
+    X_sparse = sp.csr_matrix(X)
+    base_chain = [ClassifierChain(LogisticRegression(), cv=3),
+                  RegressorChain(LinearRegression(), cv=3)]
+    [chain.fit(X_sparse, Y) for chain in base_chain]
+    for i in range(0, 2):
+        Y_pred = base_chain[i].predict(X_sparse)
+        assert_equal(Y_pred.shape, Y.shape)
+
+
+def test_base_chain_random_order():
+    # Fit base chain with random order
+    X, Y = generate_multilabel_dataset_with_correlations()
+    base_chain_random = [ClassifierChain(LogisticRegression(),
```

Comment:
I think it would be neater if you made the entire test a loop over these rather than creating a list.

Even better, use `pytest.mark.parametrize` to run the test over each estimator. Note that you won't yet find examples of this in the codebase as we don't yet use it. But you'll find plenty of examples of looping over estimators without indexing `[i]` all the time...

Reply:
I looked for the usage of pytest.mark.parametrize but could not use it here since the test runs on typing `pytest` in compiler. But I could not figure out how its going to be used here. 

Reference: https://docs.pytest.org/en/latest/parametrize.html

Reply:
Never mind parametrize. All I mean is:
```
for chain in [ClassifierChain(LogisticRegression()), RegressorChain(LinearRegression())]:
    ... fit chain or whatever
    chain_with_different_params = clone(chain).set_params(foo=bar)
```